src/Sarif/Writers/CachingLogger.cs

```diff
@@ -16,6 +17,7 @@ public class CachingLogger : BaseLogger, IAnalysisLogger
     {
         public CachingLogger(IEnumerable<FailureLevel> levels, IEnumerable<ResultKind> kinds) : base(levels, kinds)
         {
+            s_rwl = new ReaderWriterLock();
```

Comment:
>s_rwl [](http://example.com/codeflow?start=12&length=5)

you haven't articulated a high-level description of the problem and your intended solution in this PR